app/ship.py

```diff
@@ -0,0 +1,31 @@
+from __future__ import annotations
+
+
+class Ship:
+    def __init__(self, start: tuple, end: tuple) -> None:
+        self.start = start
+        self.end = end
+        self.decks_cord = self.ship_building()
+        self.kicks = set()
+
+    def ship_building(self) -> tuple:
+        if self.start[0] == self.end[0]:
+            if self.start[1] < self.end[1]:
+                return tuple(
+                    (self.start[0], i)
+                    for i in range(self.start[1], self.end[1] + 1)
+                )
+            return tuple(
+                (self.start[0], i)
+                for i in range(self.end[1], self.start[1] + 1)
+            )
+        if self.start[1] == self.end[1]:
+            if self.start[0] < self.end[0]:
+                return tuple(
+                    (i, self.start[1])
+                    for i in range(self.start[0], self.end[0] + 1)
+                )
+            return tuple(
+                (i, self.start[1])
+                for i in range(self.end[0], self.start[0] + 1)
+            )
```

Comment:
You can simplify it as follow:
[
    (i, j)
    for i in range(self.start[0], self.end[0] + 1
    for j in range(self.start[1], self.end[1] + 1
]
Try it. If it works it will be cool)

Reply:
when I wrote this, I wanted the code to be universal, regardless of whether you specify first the beginning and then the end or the end and then the beginning